src/createFormValidation.js

```diff
@@ -41,12 +41,12 @@ export function createFieldValidation(
         const value = dotProp.get(values, path)
         if (typeof value !== 'undefined' && !validate(value, values, options)) {
             let finalMessage = defaultMessage
-            if (typeof message === 'function') {
-                finalMessage = message(value, values, options)
-            }
             if (typeof message !== 'undefined') {
                 finalMessage = message
             }
+            if (typeof message === 'function') {
+                finalMessage = message(value, values, options)
+            }
```

Comment:
We need to check for the function type after checking for undefined, because otherwise, if it is a function:
1. We'll compute the message first (good)
2. We'll then also immediately overwrite that computed message with the message function (bad)

By reversing the order, we avoid this